crates/red_knot/src/symbols.rs

```diff
@@ -860,13 +880,83 @@ impl PreorderVisitor<'_> for SymbolTableBuilder {
                 ast::visitor::preorder::walk_stmt(self, stmt);
                 self.current_definition = None;
             }
+            ast::Stmt::If(node) => {
+                self.visit_expr(&node.test);
```

Comment:
Unrelated question. Does our CFG support branching inside expression or only on a statement level?

Reply:
It should support within expressions too, but it's a good callout that I should prioritize implementing a case of that to surface any issues. Also walrus expressions (within expression definitions).